internal/rbd/encryption.go

```diff
@@ -500,8 +505,13 @@ func (rv *rbdVolume) RotateEncryptionKey(ctx context.Context) error {
 		return fmt.Errorf("failed to fetch the current passphrase for %q: %w", rv, err)
 	}
 
+	// Create a new luks wrapper
+	timeoutCtx, cancel := context.WithTimeout(ctx, cryptsetup.ExecutionTimeout)
+	luks := cryptsetup.NewLUKSWrapper(timeoutCtx)
+	defer cancel()
+
```

Comment:
This should be done before obtaining locks at line 481.
Pass the same context to get/put cryptopassphrase calls as well.

Reply:
Done!